compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -316,10 +325,35 @@ pub fn push_debuginfo_type_name<'tcx>(
         ty::Param(_) => {
             output.push_str(&format!("{:?}", t));
         }
+        ty::Projection(projection_ty) => {
```

Comment:
Let me see if I can come up with an example...

Reply:
I added a test case: it comes down to referencing the associated type in a trait implementation, since the `impl$<...>` is emitted with the original definition of the generics.

Reply:
I assume this is the test case?:
```rust 
// cdb-check:[...] a!function_names::impl$<function_names::TestTrait3, GenericStruct<T, as$<T, function_names::TestTrait3>::AssocType> >::trait_function3<function_names::TestStruct1> (void)

GenericStruct::<TestStruct1, usize>::trait_function3();
```
I think we should actually generate a name like:
```rust 
// cdb-check:[...] a!impl$<function_names::TestTrait3, function_names::GenericStruct<function_names::TestStruct1, usize> >::trait_function3 (void)
```
In other words the projection should be normalized away (and the actual type should be substituted for `T`). This is how other parts of debuginfo and symbol mangling do it.